airbyte-workers/src/main/java/io/airbyte/workers/helper/ConnectionHelper.java

```diff
@@ -19,85 +14,90 @@
 import io.airbyte.config.persistence.ConfigRepository;
 import io.airbyte.scheduler.persistence.WorkspaceHelper;
 import io.airbyte.validation.json.JsonValidationException;
-import io.airbyte.workers.WorkerConfigs;
 import java.io.IOException;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.UUID;
 import lombok.AllArgsConstructor;
 
+// todo (cgardens) - we are not getting any value out of instantiating this class. we should just
+// use it as statics. not doing it now, because already in the middle of another refactor.
 @AllArgsConstructor
 public class ConnectionHelper {
```

Comment:
I am not really seeing the point of keeping this class around at all. It feels like all of the logic here should just exist in the `ConnectionsHandler` class, no?

It seems like the only thing keeping us from doing that is that the `deleteConnection` method in this class is used by the `ConnectionDeletionActivityImpl` class. Why do we need an activity to delete a connection in the first place? Could that activity use the connection API to perform this delete instead?

Reply:
the issue is that `airbyte-workers` can't depend on `airbyte-server` which is where `ConnectionsHandler` lives because it would cause a circular dependency. so there's no way to call it directly from the workflow. I think this is why this class got created in the first place, to try to get around the circular dependency.

I agree. This class probably shouldn't really exist at all and we should figure out how set up our deps so that these core data model transformations aren't in `airbyte-workers` or `airbyte-server` but their own contained module. That's a big change, so I think for now, we're making this a little better than it was but there is more to do.